### PR TITLE
Add Laravel Scout

### LINKED ISSUE #500: Integrate Meilisearch into Nova
Currently searching nova for a full name will return no results as all of the name columns are stored as parts. We did at one point have DB columns for full name but that has been moved to the PHP model and therefore isn't available for search in Nova.

### PR DESCRIPTION
**Pull request by kberzinch**
Fixes #500

### CODE REVIEW

File: app/Nova/User.php [hunk 1]
```diff
@@ -517,4 +524,75 @@ public function actions(Request $request): array
...
+
+            return $firstYear === $lastYear ? 'Member '.$firstYear : 'Member '.$firstYear.'-'.$lastYear;
+        }
+
+        $major_names = $this->majors()->pluck('display_name')->toArray();
+
+        $class_standing_names = $this->classStanding()->pluck('name')->toArray();
+
+        if (count($major_names) > 0 && null !== $major_names[0] && count($class_standing_names) > 0) {
+            return $major_names[0].' | '.ucfirst($class_standing_names[0]);
```
Comment (evan10s) [inline app/Nova/User.php:593]: Remove these pls

File: composer.json [hunk 2]
```diff
@@ -22,20 +22,24 @@
         "chelout/laravel-relationship-events": "1.4.0",
         "doctrine/dbal": "2.13.2",
         "guzzlehttp/guzzle": "7.3.0",
+        "http-interop/http-factory-guzzle": "^1.0",
         "laravel/framework": "8.50.0",
         "laravel/horizon": "5.7.9",
-        "laravel/passport": "*",
         "laravel/nova": "*",
+        "laravel/passport": "*",
```
Comment (evan10s) [inline composer.json:29]: This isn't related to your changes, but we should specify a more exact version here (the latest Passport is 10.1.3)

**Review by evan10s — COMMENTED** [review]
lgtm overall but I'd like to poke around at it in test a little bit more

File: composer.json (hunk 2 above)
Comment (JoshOldenburg) [reply composer.json:29]: Renovate will open a PR to do that after this is merged.

File: app/Nova/User.php (hunk 1 above)
Comment (kberzinch) [reply app/Nova/User.php:593]: Not necessarily opposed, but why? Is there a different value you'd like for non-members?

File: composer.json (hunk 2 above)
Comment (kberzinch) [reply composer.json:29]: I think it doesn't touch `*` packages.

File: app/Nova/User.php (hunk 1 above)
Comment (evan10s) [reply app/Nova/User.php:593]: Gah, GitHub mobile botched my comment location.  I meant the 2 `ray` lines above this

Comment (kberzinch) [reply app/Nova/User.php:593]: Fixed in c0ba56ec9a5b783245f8ee76f44098f786e01d2a

File: app/Models/Team.php [hunk 3]
```diff
@@ -88,6 +89,7 @@ class Team extends Model
     use HasRelationshipObservables;
     use Notifiable;
     use SoftDeletes;
+    use Searchable;
```
Comment (evan10s) [inline app/Models/Team.php:92]: Does this model need to have ranking rules explicitly defined?

File: app/Models/TravelAssignment.php [hunk 4]
```diff
@@ -70,6 +74,43 @@ class TravelAssignment extends Model
...
+     * The rules to use for ranking results in Meilisearch.
+     *
+     * @var array<string>
+     */
+    public $ranking_rules = [
+        'desc(user_revenue_total)',
+        'desc(user_attendance_count)',
+        'desc(user_signatures_count)',
+        'desc(user_recruiting_visits_count)',
+        'desc(user_gtid)',
```
Comment (evan10s) [inline app/Models/TravelAssignment.php:110]: Not sure but sorting by user last name might be more intuitive?  I think the GTID sort would create the effect of a seemingly random ordering by name

File: app/Models/Team.php (hunk 3 above)
Comment (kberzinch) [reply app/Models/Team.php:92]: I couldn't think of any that would be useful.

Comment (evan10s) [reply app/Models/Team.php:92]: image

It'd be nice if they appeared alphabetically

File: app/Models/TravelAssignment.php (hunk 4 above)
Comment (kberzinch) [reply app/Models/TravelAssignment.php:110]: Custom ranking rules need to be on numeric fields.

File: app/Models/Team.php (hunk 3 above)
Comment (kberzinch) [reply app/Models/Team.php:92]: Add `attendance_count`

**Review by evan10s — APPROVED** [review]
lgtm

**Review by JoshOldenburg — APPROVED** [review]

**Pull request merged**